lib/rules/no-dataset.js

```diff
@@ -0,0 +1,11 @@
+module.exports = function(context) {
+  return {
+    "Identifier": function(node) {
```

Comment:
I think we want to look for `MemberExpression`s here. `Identifier` is too generic. For an example, this would error if you called a variable `dataset`.

Theres a nice way to find what nodes you want to look for at http://eslint.org/parser/. Basically type the source of bad code and look at the AST for the closest `type`.